### ISSUE TITLE
configure with debug info (-g) without optimization

### ISSUE THREAD

**Issue by yotam-medini** [mpv-player/mpv#5472]
### mpv version and platform
0.28.0-UNKNOWN Linux 

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps
run under gdb

### Expected behavior
If I could have compilation without optimization and with -g, then
stepping in gsb would consistently step  forward and never give 
"optimized out" when trying to print variable values.

### Actual behavior
In gdb - frequent jumping backward in code, and showing some variables
as "optimized out"

### Log file

Make a log file made with -v or --log-file=output.txt, paste it to
http://sprunge.us or a similar site, and replace this text with a link to it.

Providing a log file is strongly encouraged. It is very helpful for reproducing
bugs or getting important technical context. Without a log it might not be
possible to analyze and fix certain bugs, and your bug report will be closed
or ignored.

### Sample files

Sample files needed to reproduce this issue can be uploaded to https://0x0.st/
or similar sites. (Only needed if the issue cannot be reproduced without it.)
Do not use garbage like "cloud storage", especially not Google Drive.

I am new to 'waf' flavored build tool.
How can I see and modify its hidden CFLAGS, and gcc compilation  command line ?
Basically, I wuld like to have -g and remove all -Oxxx optimizations.

**Comment by sfan5** [comment]
configure with `CFLAGS="-g2 -O1" ./waf configure`
This is not a bug in mpv.

**Comment by wm4** [comment]
There is a configure flag to disable optimizations.

**Comment by yotam-medini** [comment]
For debugging, I  tried configuring with
   `CFLAGS="-g2 -O0" ./waf configure --disable-gl`
then while running 
  ` ./waf`
the
   `ps laxaww`
command  showed (among many similar compilation processes):

    0 1000 30283 30282 20 0 55072 25432 - R pts/10 0:00
    /usr/lib/gcc/x86_64-linux-gnu/5/cc1 -quiet -I . -I .. -I
    /usr/local/include -I /usr/include/alsa -I /usr/include/harfbuzz -I
    /usr/include/freetype2 -I /usr/include/fribidi -imultiarch
    x86_64-linux-gnu -D_REENTRANT -D _ISOC99_SOURCE -D _GNU_SOURCE -D
    _LARGEFILE_SOURCE -D _FILE_OFFSET_BITS=64 -D _LARGEFILE64_SOURCE
    ../audio/audio.c -quiet -dumpbase audio.c -mtune=generic -march=x86-64
    -auxbase-strip /home/yotam/pub/mpv/build/audio/audio.c.19.o -g2 -g -O0 -O2 
     -Werror=implicit-function-declaration
    -Wno-error=deprecated-declarations -Wno-error=unused-function
    -Wempty-body -Wdisabled-optimization -Wstrict-prototypes
    -Wno-format-zero-length -Werror=format-security -Wno-redundant-decls
    -Wvla -Wall -Wundef -Wmissing-prototypes -Wshadow -Wno-switch
    -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result
    -std=c99 -fstack-protector-strong -Wformat-security -o /tmp/ccaRO38d.s

How can I get rid of the '-O2' optimization from the  `-g2 -g -O0 -O2 ` options?

**Comment by CounterPillow** [comment]
`./waf --disable-optimize`

**Comment by yotam-medini** [comment]
Thanks!  This option should be used at configure phase, namely:
`CFLAGS="-g2 -O0" ./waf configure --disable-optimize
`
